virtcontainers/qemu.go

```diff
@@ -837,6 +853,64 @@ func (q *qemu) hotplugRemoveCPUs(amount uint32) (uint32, error) {
 	return amount, q.sandbox.storage.storeHypervisorState(q.sandbox.id, q.state)
 }
 
+func (q *qemu) hotplugMemory(slot uint32, memMB uint64, op operation) error {
+	// We do not support memory hot unplug.
+	if op == removeDevice {
+		return errors.New("cannot hot unplug memory device")
+	}
+
+	// rule out extremely large parameter
+	if memMB > math.MaxInt32 {
+		return fmt.Errorf("cannot hotplug %d MB memory: size too large", memMB)
+	}
+
+	maxMem, err := q.hostMemMB()
+	if err != nil {
+		return err
+	}
+
+	// calculate current memory
+	currentMemory := int(q.config.DefaultMemSz)
```

Comment:
This configuration is the one that comes from `kata-containers.toml` if that is the case this may change if the user changes the value. In that case current memory wont be the same memory that we used to boot.

If that is the case we have a few options here:

- When we start the VM save the memory that was used to start it.
- We can query to qmp to get the current memory in the guest.

Reply:
I think saving the amount of memory used when the VM was started is the best option here. Or we could simply save the current amount of memory plugged to the VM (but this is because we cannot unplug that I am thinking about this).

Reply:
@jcvenegas 
> This configuration is the one that comes from kata-containers.toml if that is the case this may change if the user changes the value. In that case current memory wont be the same memory that we used to boot.

Sorry, I don't quit follow it. Do you mean the sandbox config we used to boot can change later when someone connects to the sandbox again? It doesn't sound like a solid solution. But I think we only use `kata-containers.toml` when creating the sandbox and later access to the sandbox all fetches the sandbox config from persistent metadata with the sandbox ID. Am I missing something?

Reply:
@bergwolf yes, well I am considering a future case were someone want to update the memory in a container. And we want to increase the memory. In that case exist the case that the config file was modified.

Another case is when we create a container inside a Pod. In that case the pod may be created with one config file, but then may be updated with a different default memory, and then create a container inside a pod. 

But agree for the current use case does not have any issue with that.